src/csync/csync_exclude.cpp

```diff
@@ -204,16 +206,35 @@ bool csync_is_windows_reserved_word(const char* filename) {
             return true;
         }
     }
-    if (c_strncasecmp(filename, win_reserved_words[j], len_reserved_word) == 0) {
-        if (len_filename == len_reserved_word) {
-            return true;
+
+    if (len_filename == 3 || (len_filename > 3 && filename[3] == '.')) {
+        size_t n_words = sizeof(win_reserved_words_3) / sizeof(char *);
+        for (size_t j = 0; j < n_words; j++) {
+            if (c_strncasecmp(filename, win_reserved_words_3[j], 3) == 0) {
+                return true;
+            }
+        }
+    }
+
+    if (len_filename == 4 || (len_filename > 4 && filename[4] == '.')) {
+        size_t n_words = sizeof(win_reserved_words_4) / sizeof(char *);
+        for (size_t j = 0; j < n_words; j++) {
+            if (c_strncasecmp(filename, win_reserved_words_4[j], 4) == 0) {
+                return true;
+            }
         }
-        if ((len_filename > len_reserved_word) && (filename[len_reserved_word] == '.')) {
+    }
+
+    size_t n_words = sizeof(win_reserved_words_n) / sizeof(char *);
+    for (size_t j = 0; j < n_words; j++) {
+        const char *word = win_reserved_words_n[j];
+        size_t len_word = strlen(word);
+        if (len_word == len_filename && c_strncasecmp(filename, word, len_word) == 0) {
             return true;
         }
     }
-  }
-  return false;
+
+    return false;
 }
 
 static CSYNC_EXCLUDE_TYPE _csync_excluded_common(c_strlist_t *excludes, const char *path, int filetype, bool check_leading_dirs) {
```

Comment:
My benchmark test shows that this is still significant when there are no exclude.  Can they not be put in the regexp?

Reply:
I don't understand what you mean? That excluded_common still shows up in benchmarks even when there are no excludes?